bin/lang-js/BUCK

```diff
@@ -1,113 +1,84 @@
 load(
     "@prelude-si//:macros.bzl",
-    "eslint",
+    "alias",
     "export_file",
-    "ts_test",
     "nix_omnibus_pkg",
-    "node_pkg_bin",
-    "npm_bin",
-    "package_node_modules",
-    "prettier_check",
-    "typescript_check",
-    "typescript_dist",
-    "typescript_runnable_dist",
-    "typescript_runnable_dist_bin",
 )
 load(
-    "@prelude-si//:pnpm.bzl",
-    "pnpm_task_library",
-    "pnpm_task_binary",
-    "pnpm_task_test",
+    "@prelude-si//:deno.bzl",
+    "deno_compile",
+    "deno_format",
+    "deno_test",
 )
 
 export_file(
     name = "package.json",
 )
 
-package_node_modules(
-    name = "node_modules",
-)
-
-filegroup(
-    name = "src",
-    srcs = glob([
-        "src/**/*.ts",
-        "tsconfig.json",
-    ]),
-)
-
-filegroup(
-    name = "test_src",
-    srcs = glob([
-        "tests/**/*",
-    ]),
-)
-
-prod_deps_srcs = {
-    "lib/ts-lib": "//lib/ts-lib:src",
-}
-
-dev_deps_srcs = {
-    "lib/eslint-config": "//lib/eslint-config:src",
-    "lib/tsconfig": "//lib/tsconfig:src",
-}
-
-npm_bin(
-    name = "tsup",
-)
-
-typescript_dist(
-    name = "dist",
-    srcs = [":src"],
-    tsup = ":tsup",
-    prod_deps_srcs = prod_deps_srcs,
-    dev_deps_srcs = dev_deps_srcs,
+export_file(
+    name = "deno.json",
 )
 
-typescript_runnable_dist(
+alias(
     name = "lang-js",
+    actual = ":bin"
 )
 
-typescript_runnable_dist_bin(
+deno_compile(
     name = "bin",
-    typescript_runnable_dist = ":lang-js",
+    main = "src/index.ts",
+    out = "lang-js",
+    srcs = glob([
+        "src/**/*.ts",
+        "src/**/*.js",
+    ]) +
+    ["//lib/ts-lib-deno:ts-lib-deno"],
+    permissions = [
+        "allow-all",
+    ],
+    unstable_flags = [
+        "worker-options",
+    ],
+    visibility = ["PUBLIC"],
 )
 
-eslint(
-    name = "check-lint",
-    directories = ["src", "tests"],
-    srcs = [":src", ":test_src"] + glob([".eslint*"]),
-    prod_deps_srcs = prod_deps_srcs,
-    dev_deps_srcs = dev_deps_srcs,
+deno_format(
+    name = "fix-format",
+    srcs = glob(["**/*.ts", "**/*.js"]),
+    ignore = ["node_modules"],
 )
 
-typescript_check(
-    name = "check-type",
-    srcs = [":src"],
-    prod_deps_srcs = prod_deps_srcs,
-    dev_deps_srcs = dev_deps_srcs,
+deno_format(
+    name = "check-format",
+    srcs = glob(["**/*.ts", "**/*.js"]),
+    check = True,
 )
```

Comment:
Eventually I think we could push the default declarations of these rules into the `deno_compile` *macro* definition, similar to our `rust_binary`/`rust_library` rules (they help to define format checks/fixes, unit tests, etc). That can be a later refactoring though (and would be more useful for our second/third Deno sub-project).